### PR TITLE
Added code to prevent entering past dates for jobs and events

### PR DESCRIPTION
**Pull request by jpc11623**

### CODE REVIEW

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
         var vm = this;
+        vm.invalidDate = false;
+        vm.currentDate = new Date(Date.now());
         vm.selectedItem = {};
         vm.postType = $stateParams.type.toUpperCase();
         vm.postTypeValues = CorkboardPostTypes[vm.postType];
         //Refactor for style and TEST
         vm.submitPost = function() {
-            if ($scope[vm.postTypeValues.form].$invalid) {
+            if ($scope[vm.postTypeValues.form].$invalid || vm.invalidDate === true) {
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Do we need the vm.invalidDate here. I think the $scope[vm.PostTypeValues.form] check the entire state of the form.

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 2]
```diff
@@ -6,12 +6,15 @@
...
+                debugger;
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove the debugger statements

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 3]
```diff
@@ -47,5 +50,33 @@
...
+        vm.validateDate = function(date) {
+            var validatingDate = new Date(date);
+            validatingDate.setHours(0,0,0,0);
+
+            debugger;
+            vm.currentDate.setHours(0,0,0,0);
+
+            if (date){
+                if (validatingDate < vm.currentDate) {
+                    debugger;
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove the debugger statements

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 4]
```diff
@@ -47,5 +50,33 @@
...
+                    vm.invalidDate = true;
+                    //vm.eventForm.Date.$error = { date: true };
+                } else {
+                    debugger;
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove the debugger statements

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 5]
```diff
@@ -47,5 +50,33 @@
...
+                    vm.invalidDate = false;
+                }
+
+            }
+            else {
+                debugger;
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove the debugger statements

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 6]
```diff
@@ -47,5 +50,33 @@
...
         });
+
+        vm.validateDate = function(date) {
+            var validatingDate = new Date(date);
+            validatingDate.setHours(0,0,0,0);
+
+            debugger;
+            vm.currentDate.setHours(0,0,0,0);
+
+            if (date){
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Think about simplifying the if then else if logic using guard statements

File: website/app/corkboard/templates/post-event.html [hunk 7]
```diff
@@ -19,11 +19,11 @@ <h1 class="page-header">Post an Event</h1>
         </ng-messages>
       </div>
     </div>
-    <div class="form-group" ng-class="{'has-error': (eventForm.Date.$pristine || eventForm.Date.$invalid) && eventForm.$submitted}">
+    <div class="form-group" ng-class="{'has-error': ((eventForm.Date.$pristine || eventForm.Date.$invalid) && eventForm.$submitted) || corkboardEdit.validateDate(corkboardEdit.selectedItem.Date) }">
```
Comment (charlie-retzler) [inline website/app/corkboard/templates/post-event.html]: Is this necessary or is the eventForm.Date.$invalid also handling this?

File: website/app/corkboard/controllers/corkboard.edit.controller.js (hunk 1 above)
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Is InvalidDate check necessary here. Same reasoning as above

File: website/app/corkboard/controllers/corkboard.edit.controller.js [hunk 8]
```diff
@@ -47,5 +50,33 @@
                 }
             }
         });
+
+        vm.validateDate = function(date) {
+            var validatingDate = new Date(date);
+            validatingDate.setHours(0,0,0,0);
+
+            debugger;
```
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove debugger statement

File: website/app/corkboard/controllers/corkboard.edit.controller.js (hunk 3 above)
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove debugger

File: website/app/corkboard/controllers/corkboard.edit.controller.js (hunk 5 above)
Comment (charlie-retzler) [inline website/app/corkboard/controllers/corkboard.edit.controller.js]: Remove debugger

File: website/app/corkboard/templates/post-event.html [hunk 9]
```diff
@@ -31,13 +31,18 @@ <h1 class="page-header">Post an Event</h1>
         <ng-messages for="eventForm.Date.$error" ng-if="eventForm.$submitted">
           <ng-messages-include src="on-submit-messages"></ng-messages-include>
         </ng-messages>
+        <ng-messages for="corkboardEdit.invalidDate" ng-if="corkboardEdit.invalidDate" src="on-error-messages">
+          <ng-messages-include src="on-submit-messages">
+            <span ng-bind-html="$root.messages[$root.CORKBOARD_MESSAGES.corkboardDateError].message | html">aaa</span>
```
Comment (charlie-retzler) [inline website/app/corkboard/templates/post-event.html]: Should aaa be the content of the span?

File: website/app/corkboard/templates/post-job.html [hunk 10]
```diff
@@ -24,10 +24,10 @@ <h1 class="page-header">Post a Job</h1>
         </ng-messages>
       </div>
     </div>
-    <div class="form-group" ng-class="{'has-error': (jobForm.StartDate.$pristine || jobForm.StartDate.$invalid) && jobForm.$submitted}" ng-show="corkboardEdit.selectedItem.StartDateQualifier == 'Before' || corkboardEdit.selectedItem.StartDateQualifier == 'After'">
+    <div class="form-group" ng-class="{'has-error': ((jobForm.StartDate.$pristine || jobForm.StartDate.$invalid) && jobForm.$submitted) || corkboardEdit.validateDate(corkboardEdit.selectedItem.StartDate)}" ng-show="corkboardEdit.selectedItem.StartDateQualifier == 'Before' || corkboardEdit.selectedItem.StartDateQualifier == 'After'">
```
Comment (charlie-retzler) [inline website/app/corkboard/templates/post-job.html]: Do we need to call out corkboardEdit.validateDate(corkboardEdit.selectedItem.StartDate) or can this be part of the StartDate.$invalid?

File: website/app/corkboard/templates/post-job.html [hunk 11]
```diff
@@ -35,6 +35,11 @@ <h1 class="page-header">Post a Job</h1>
         <ng-messages for="jobForm.StartDate.$error" ng-if="jobForm.$submitted">
           <ng-messages-include src="on-submit-messages"></ng-messages-include>
         </ng-messages>
+        <ng-messages for="corkboardEdit.invalidDate" ng-if="corkboardEdit.invalidDate" src="on-error-messages">
+          <ng-messages-include src="on-submit-messages">
+            <span ng-bind-html="$root.messages[$root.CORKBOARD_MESSAGES.corkboardDateError].message | html">aaa</span>
```
Comment (charlie-retzler) [inline website/app/corkboard/templates/post-job.html]: Is aaa needed in the span?

File: website/app/corkboard/templates/post-need.html [hunk 12]
```diff
@@ -10,7 +10,7 @@ <h1 class="page-header">Post a Need</h1>
         </ng-messages>
       </div>
     </div>
-    <div class="form-group" ng-class="{'has-error': (editPost.Description.$pristine || editPost.Description.$invalid) && editPost.$submitted }">
+    <div class="form-group" ng-class="{'has-error': (editPost.Description.$pristine || editPost.Description.$invalid) && editPost.$submitted && editPost }">
```
Comment (charlie-retzler) [inline website/app/corkboard/templates/post-need.html]: Should the change on this line be something more than "&& editPost"?

File: website/spec/controllers/corkboard.edit.controller.spec.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
+describe('CorkboardEditController', function() {
+
+    beforeEach(module('crossroads.corkboard'));
+
+    /*var mockAdminRootScope = getAdminRootScope();
```
Comment (charlie-retzler) [inline website/spec/controllers/corkboard.edit.controller.spec.js]: Remove commented out code

**Comment by IngageKrizJ** [comment]
Looks good to me.

**Pull request merged**